core/src/main/java/io/grpc/ControlPlaneScheduler.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2018 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc;
+
+import java.util.concurrent.TimeUnit;
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * Defines operation scheduling services for control plane components.  Assuming all control-plane
+ * operations are serialized by some mechanism (e.g., the "Channel Executor" mentioned in {@link
+ * LoadBalancer} API, this scheduler guarantees that all runnables are run in the synchronization
+ * context of a certain control plane component.
+ */
+@ThreadSafe
+@ExperimentalApi("TODO")
+public abstract class ControlPlaneScheduler {
+  /**
+   * Schedules a task to run as soon as poassible.
+   *
+   * <p>Non-reentrency is guaranteed.  If this method is called from within an operation in the
+   * synchronization context, the task will be queued and run after the current operation has
+   * finished, rather than running inline.
+   *
+   * <p>Ordering is guaranteed.  Tasks are guaranteed to run in the same order as they are
+   * submitted.
+   */
+  public final ScheduledContext scheduleNow(Runnable task) {
+    return schedule(task, 0, TimeUnit.NANOSECONDS);
```

Comment:
I'm not excited about making an easy way to make a zero-delay task. This just abuses the scheduled executor and is a strong code smell to me.

Reply:
It's the same as the current `runSerialized()`. I still don't understand the issue.

Reply:
I'm not sure what you're saying is the same. Today runSerialized() runs on the current thread:
https://github.com/grpc/grpc-java/blob/v1.15.0/core/src/main/java/io/grpc/internal/ManagedChannelImpl.java#L1236-L1238

And any schedule() would run on a separate thread. I'm against having schedule() turn into running on the current thread based on the timeout.

Reply:
Fair enough. I have decoupled `schedule()` with `scheduleNow()`.